paimon-core/src/main/java/org/apache/paimon/schema/SchemaValidation.java

```diff
@@ -144,6 +145,12 @@ public static void validateTableSchema(TableSchema schema) {
                             + "still want to keep the primary key definition.");
         }
 
+        if (options.bucket() == -1
+                && schema.primaryKeys().isEmpty()
+                && options.toMap().get(BUCKET_KEY.key()) != null) {
+            throw new RuntimeException("Cannot define 'bucket-key' in unaware bucket mode.");
```

Comment:
remove  `schema.primaryKeys().isEmpty()` and here `in unaware or dynamic bucket mode`.